[15.0][MOV] rename mrp_production_cancel_manager to mrp_production_buttons_manager

IriaAlonso opened the pull request:
- Rename mrp_production_cancel_manager to mrp_production_buttons_manager
- Generalize deny access to buttons to Non-Managers like "scrap", "unreserve", "check availability" and "mark as done"

@dalonsod

dalonsod commented on mrp_production_buttons_manager/views/mrp_production_view.xml:
```diff
@@ -0,0 +1,36 @@
+            <xpath expr="//button[@name='action_cancel'][2]" position="attributes">
+                <attribute name="groups">mrp.group_mrp_manager</attribute>
+            </xpath>
+            <xpath expr="//button[@name='button_mark_done'][2]" position="attributes">
```
This is weird... why start for `button_mark_done` second appearance in view?

IriaAlonso commented:
@dalonsod done

dalonsod commented:
Git history for the most files will be properly reserved, but not for these two ones:

![imagen](https://github.com/user-attachments/assets/b63b3733-500c-4d50-9818-670311b8eae3)

Could you check?

IriaAlonso committed:
[MOV] rename mrp_production_cancel_manager to mrp_production_buttons_manager

IriaAlonso committed:
[IMP] mrp_production_buttons_manager: added new buttons to restrictions

With this improvement only managers can access to the following buttons:
- Mark as done
- Scrap
- Unreserve
- Check Availability

And "Cancel" button limited access is kept as well.

dalonsod reviewed (approved):
Well keep this operations splitted in two commits, :+1: 

I've only improved second commit title and detailed description, in order to make it more descriptive, please take a look @IriaAlonso 

Merging!!!
